src/helpers/cacheUtils.ts

```diff
@@ -0,0 +1,46 @@
+// (C) 2007-2018 GoodData Corporation
+import set = require('lodash/set');
+import { SDK, factory as createSdk } from '@gooddata/gooddata-js';
+
+const gdcCache = {};
+
+export const FEATURE_FLAGS = 'featureFlags';
+export const COLOR_PALETTE = 'colorPalette';
+
+async function isColorPaletteEnabled(cache: any, projectId: string, sdk: SDK) {
+    let featureFlags;
+
+    // tslint:disable-next-line:prefer-conditional-expression
+    if (cache[projectId] && cache[projectId][FEATURE_FLAGS]) {
+        featureFlags = await cache[projectId][FEATURE_FLAGS];
+    } else {
+        featureFlags = await sdk.project.getFeatureFlags(projectId);
+    }
+    return featureFlags.isColorPaletteEnabled;
+}
+
+function isContentInCache(cache: any, projectId: string, contentType: string) {
+    return cache[projectId] && cache[projectId][contentType];
+}
+
+export async function getCachedData(projectId: string, contentType: string, sdk: SDK = createSdk()) {
+    if (isContentInCache(gdcCache, projectId, contentType)) {
+        return gdcCache[projectId][contentType];
```

Comment:
I would prefer having the cache as 
```
{
featureFlags: { [pid]: flags},
colorPalette: { [pid]: flags},
}
```

Why? Because if some other setting will not be project-based (user-based or whatever), we would be mixing this info on the top-level.

Reply:
I would suggest that cache key might be different for some settings:
```
{
  [`PROJECT_${projectId}`]: { featureFlags: ... },
  [`USER_${userId}`]: { userSettings: ... },
  [`USER_${userId}_PROJECT_${projectId}]
}
```
Prefixing keys by type helps with duplicity between userIds and projectIds

Then we need more than one function:
`getCachedProjectData(projectId, contentType, sdk)`
`getCachedUserData(userId, contentType, sdk)`
`getCachedUserProjectData(userId, projectId, contentType, sdk)`